frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -248,6 +248,31 @@ function ReaderHighlight:onTapXPointerSavedHighlight(ges)
     end
 end
 
+function ReaderHighlight:updateHighlight(page, index, move_from_beginning, dir)
+    local highlight = self.view.highlight.saved[page][index]
+    local highlight_beginning = highlight.pos0
+    local highlight_end = highlight.pos1
+    if move_from_beginning then
+        if dir == 1 then
+            local updated_highlight = self.ui.document:getNextVisibleWordEnd(highlight_beginning)
+            self.view.highlight.saved[page][index].pos0 = updated_highlight
+        else
```

Comment:
Might be easier to read with both args as numbers (`0` = use `pos0`, `1` = use `pos1`) and direction (-1 backward, 1 forward as you did)
```lua
function ReaderHighlight:updateHighlight(page, index, side, direction)
    local highlight = self.view.highlight.saved[page][index]
    if side == 0 then
        if direction == 1 then
            local new_pos0 = self.ui.document:getNextVisibleWordEnd(highlight.pos0)
            if new_pos0 then -- just in case to avoid side effects, as the cre.cpp can return nil (its return 0) if the xpointer is not found, which might happen, you never know
                self.view.highlight.saved[page][index].pos0 = new_pos0
            end
        else
```